lib/User/Backend.php

```diff
@@ -229,6 +231,24 @@ public function getCurrentUserId(): string {
 			}
 		}
 
+		// check if we should ask the OIDC Identity Provider app (app_id: oidc) to validate the token (default is false)
+		if ($ncOidcProviderBearerValidation) {
+			if (class_exists('OCA\OIDCIdentityProvider\Event\TokenValidationRequestEvent')) {
+				$validationEvent = new \OCA\OIDCIdentityProvider\Event\TokenValidationRequestEvent($headerToken);
+				$this->eventDispatcher->dispatchTyped($validationEvent);
```

Comment:
Since this calls external code we cannot be sure about that it remains unchanged. I'm not a fan API wise but cannot think of a better way.

I would wrap this with try/catch generally on any Throwable to avoid bringing down an instance where any issue occurs here